go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -311,33 +309,35 @@ func nadNeedsUpdate(oldNAD, newNAD *nettypes.NetworkAttachmentDefinition) bool {
 		return false
 	}
 
-	return !reflect.DeepEqual(oldNAD.Spec, newNAD.Spec)
+	// also reconcile the network in case its route advertisements changed
+	return !reflect.DeepEqual(oldNAD.Spec, newNAD.Spec) ||
+		oldNAD.Annotations[types.OvnRouteAdvertisementsKey] != newNAD.Annotations[types.OvnRouteAdvertisementsKey]
```

Comment:
wait we are loading the route advertisement that applies to this NAD as an annotation on the NAD? That seems backwards. It seems like we would want to set a status on the route advertisement CRDs for selected+active networks.

Reply:
With what purpose? Those are two different things: the annotation is just an implementation detail (more less) while adding to status is an API. I don't wanna be the first one to post on status selected things by a selector, I haven't seen any API doing that, could turn out to be either long lists, lots of updates or both. Or do you see a functional problem here that would be avoided?

Reply:
An example would be with CUDN, where we (are supposed to) post each namespace selected and if its network is setup/active.

I was thinking that we could centralize the information into the RA CR so then both the user and ovnk would have a single source for that information. I don't see a functional problem with using the annotation, but I fear if this is the method we use, then in the future many controllers will be trying to store annotations to the NAD as their source of truth for the state of the network. This could cause performance issues, and it seems better imo to go with the status in the CR itself.

I wont block this PR on it, its something we can patch later since right now its only an internal technical detail.